### DISCUSSION TITLE
[FeatureRequest] Provisioning: Allow provisioning Service Accounts (with API keys)

### DISCUSSION THREAD

**Discussion by pdf** [grafana/grafana#49552] [General]
Service Accounts offer a nice method of integrating external tooling with Grafana without relying on authentication as a particular user, however the story for automated deployment is problematic.

Currently to create a service account via automation, it appears that the only method is via API calls, which requires authentication via an existing user/key, which defeats the purpose of service accounts - we're back to relying on a particular user's credentials being available for the action.

It also requires some not-insignificant plumbing - writing a client that does a multi-stage dance through the API to:

1. Check for an existing SA
2. Create the SA
3. Create an API key for the SA
4. Export the SA ID and API key for use

If I was doing this in Kubernetes, for example, I'd probably have to resort to writing a whole operator or something, backed by user credentials, to make this work with GitOps.

I propose that there should be a filesystem-backed method to provision service accounts, and associated pre-defined API keys, so that user accounts and API round-trips do not need to be involved in the process.

**Comment by pawelros** [comment]
I have exactly the same problem. We have a GitHub actions that adds grafana annotation for each deployment. Whenever we re-deploy grafana or restart pods in k8s our API key disappears and we need to use horrible hacks to generate a new API KEY and update it in GitHub. We need a way to provision service accounts with predefined API KEY to make our lives much easier.

**Comment by Hronom** [comment]
Same here, very needed feature. 
Currently we need it in order to create grafana setup with pre-installed API key to upload dashboards/alters etc.

Grafana already allow to provision password for admin so we need same for service account.

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/identity-access-team

**Comment by Christian-Little-Tava** [comment]
https://github.com/grafana/grafana/issues/82987
For others who wind up here from Google
